characteristic/characteristic_test.go

```diff
@@ -5,6 +5,36 @@ import (
 	"testing"
 )
 
+func TestCharacteristicGetValue(t *testing.T) {
+	i := 0
+
+	c := NewCharacteristic(TypeOn)
+	c.Value = i
+
+	c.OnValueUpdateFromConn(func(conn net.Conn, c *Characteristic, new, old interface{}) {
```

Comment:
We should check if `OnValueUpdateFromConn` is ever called by updating a counter inside the function. For example

```go
callCount := 0

c.OnValueUpdateFromConn({
    callCount += 1
})

if is, want := callCount, 1; is != want { ... }
...
if is, want := callCount, 2; is != want { ... }
...
```